### DISCUSSION TITLE
bluetooth delay

### DISCUSSION THREAD

**Discussion by simon-auch** [rern/rAudio#784] [Fixed Issues]
When watching some videos on my phone (youtube, using newpipe for example) and sending the audio to rAudio over bluetooth there is a noticable delay of around ~1 second which, for me, is not acceptable. 
My setup is basically phone -> bluetooth -> rAudio (rpi zero 2w) -> dac. One could also argue that the dac adds a lot of delay, but I'm not sure how exactly I could test this.

Are there any recommendations on how to reduce the audio delay?

**Comment by ElliottBridge** [comment]
the delay from the dac should be less than 200ms (max), there must be something up with background processing.

**Reply by simon-auch** [reply]
I'm not aware of having any background processing enabled on the raspberry, this problem already appears with a fresh install of rAudio-1

**Comment by Slaughthammer** [comment]
I guess it's just some crazy long buffers in bluez-alsa to get rid of dropouts that typically occur when using the RasPi's WiFi and bluetooth at the same time. However, Id' also be interested in getting close to real time (a couple of 10 ms delay maybe) audio via bluetooth. But I fear that only works if WiFi is disabled or a separate adapter is used.

**Comment by simon-auch** [comment]
Small update after I finally got around to it:
I setup a basic arch install on the rpi, using pipewire as the audio system. With that I managed to get acceptable delay (not noticeable when watching movies, etc). This however only worked when disabling wifi (with wifi enabled there was constant crackling) a seperate bluetooth adapter would probably fix this however.
